src/Illuminate/Collections/Collection.php

```diff
@@ -939,13 +950,24 @@ public function search($value, $strict = false)
     }
 
     /**
-     * Get and remove the first item from the collection.
+     * Get and remove the first N items from the collection.
      *
+     * @param  int  $count
      * @return mixed
      */
-    public function shift()
+    public function shift($count = 1)
     {
-        return array_shift($this->items);
+        if ($count === 1) {
+            return array_shift($this->items);
+        }
+
+        $results = [];
+
+        foreach (range(1, $count) as $item) {
+            array_push($results, array_shift($this->items));
```

Comment:
`array_shift` is an expensive operation (as it needs to re-key all items in the array). You don't really want to do that in a loop 😢